src/txn/txn.c

```diff
@@ -1219,10 +1219,13 @@ __txn_resolve_prepared_op(WT_SESSION_IMPL *session, WT_TXN_OP *op, bool commit,
           txn->id, __wt_timestamp_to_string(txn->prepare_timestamp, ts_string[0]));
 
     /*
-     * Aborted updates can exist in the update chain of our transaction. Generally this will occur
-     * due to a reserved update. As such we should skip over these updates.
+     * Aborted updates can exist in the update chain of our transaction due to reserved update. All
+     * the prepared updates on a key by a transaction will be resolved during the resolution of the
+     * first operation on that key. Hence the update chain could contain the prepared updates by
```

Comment:
So basically the update chain has transitioned to the committed state, then another transaction has written to it concurrently which is why we see updates from a different txn on the chain?

Reply:
Yes.